lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2023,6 +2023,19 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     return BM->addBinaryInst(OpFAdd, Ty, Mul,
                              transValue(II->getArgOperand(2), BB), BB);
   }
+  case Intrinsic::usub_sat: {
+    // usub.sat(a, b) -> (a > b) ? a - b : 0
+    SPIRVType *Ty = transType(II->getType());
+    Type *BoolTy = IntegerType::getInt1Ty(M->getContext());
+    SPIRVValue *Sub =
+        BM->addBinaryInst(OpISub, Ty, transValue(II->getArgOperand(0), BB),
+                          transValue(II->getArgOperand(1), BB), BB);
+    SPIRVValue *Cmp = BM->addCmpInst(
+        OpUGreaterThan, transType(BoolTy), transValue(II->getArgOperand(0), BB),
+        transValue(II->getArgOperand(1), BB), BB);
```

Comment:
Minor: would it be better to store translated arguments to local variables to avoid calling `transValue` twice for the same value?

Note: this is not a performance improvement as `transValue` already has a cache of translated values, this is just code style question

Reply:
Okay, done.